opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -0,0 +1,724 @@
+                    ]:
+                        batch_q.append(example.texts[1])
+                        batch.append(example.texts[0])
+
+                    out_q = model.tokenize(batch_q)
+                    for key in out_q.keys():
+                        out_q[key] = out_q[key].to(device)
+                    Y = model(out_q)["sentence_embedding"]
+
+                    out = model.tokenize(batch)
+                    for key in out.keys():
+                        out[key] = out[key].to(device)
+                    X = model(out)["sentence_embedding"]
+
+                    XY = torch.exp(torch.matmul(X, Y.T) / (X.shape[1]) ** 0.5)
+                    num = torch.diagonal(XY)
+
+                    den0 = torch.sum(XY, dim=0)
+                    den1 = torch.sum(XY, dim=1)
+
+                    train_loss = -torch.sum(torch.log(num / den0)) - torch.sum(
+                        torch.log(num / den1)
+                    )
+                    loss.append(train_loss.item())
+                    train_loss.backward()
+                    optimizer.step()
+                    scheduler.step()
+                    optimizer.zero_grad()
+
+                    if not j % 500 and j != 0:  # MS: change 500
+                        plt.plot(loss[::100])
+                        plt.show()
+
+        # saving the pytorch model and the tokenizers.json file is saving at this step
+        model.save("trained_pytorch_model/")
+        device = "cpu"
+        cpu_model = model.to(device)
+        print(f"Total training time: {time.time() - init_time}")
+
+        for key in out_q.keys():
+            out_q[key] = out_q[key].to(device)
+
+        traced_cpu = torch.jit.trace(
+            cpu_model,
+            (
+                {
+                    "input_ids": out_q["input_ids"],
+                    "attention_mask": out_q["attention_mask"],
+                }
+            ),
+            strict=False,
+        )
+
+        print("Preparing model to save")
+        torch.jit.save(traced_cpu, output_model_path)
+
+        print("Model saved to path: " + output_model_path)
+        return model
+
+    def zip_model(self, model_path: str = None, zip_file_name: str = None) -> None:
+        """
+        Description:
+        zip the model file and its tokenizer.json file to prepare to upload to the Open Search cluster
+
+        Parameters:
+        model_path: str
+            Optional, path to find the model file, if None, default as trained_model.pt file in current path
+        zip_file_name: str =None
+            Optional, file name for zip file. if None, default as zip_model.zip
+
+        Return:
+            None
+        """
+
+        if model_path is None:
+            model_path = os.path.join(os.getcwd(), "trained_model.pt")
+
+        if zip_file_name is None:
+            zip_file_name = "zip_model.zip"
+
+        if not os.path.exists("trained_pytorch_model/tokenizer.json"):
```

Comment:
I think we talked about putting these (trained_pytorch_model/tokenizer.json) in a static variable for reuse? Same for others like: `trained_model.pt`

Reply:
We did talk about adding static variable and class variables in the future commit, yes